src/action_install.rs

```diff
@@ -236,3 +230,103 @@ pub fn check_tars_and_move(name: &str, dirs: &RuaDirs, archive_whitelist: &Index
 		});
 	}
 }
+
+/// Generates the dependencies map of 1st and 2nd depth lvl of the
+/// package storing also if the dependencies are from aur or not.
+fn gen_package_deps_map(
+	map: &mut HashMap<String, Vec<(String, String)>>,
+	dep_name: &str,
+	alpm: &alpm::Alpm,
+) {
+	let (_, pacman_deps, aur_deps) =
+		aur_rpc_utils::recursive_info(&[dep_name.to_string()], alpm).unwrap();
```

Comment:
Hi. I've been playing around with the code a bit more. I noticed that we're calling `aur_rpc_utils::recursive_info` here. Is it possible to avoid doing that? The problem with calling it is that it's an actual HTTP call (or many HTTP calls) that need to be made, which adds additional slow-down, and pushes you towards RPC limits faster. Technically, the topmost call in `install` method already gives you all the information that you'll need in future. The question is only how to aggregate and use the already fetched information.

BTW, I'm OK to pick the tree-printing code for working as well (I'd even do the full tree), so if you feel tired of dealing with it, I can take over. If you'd like to push it to the successful end then it's fine as well of course.

Reply:
And sorry again for the late response - only back in Amsterdam this week after my vacation.

Reply:
The problem is easy. 
The info about depencendies that `aur_rpc_utils::recursive_info()` provides is not enough.
If we want the deps of the deps or "2nd order dependencies" we need the `Package` struct of the `1st order depencencies`, but with the `aur_rpc_utils::recursive_info()` call we only get the names as `String`. So to get the `2nd order deps` I need to call `aur_rpc_utils::recursive_info()` for each "1st order dependency" to get the next ones.

I don't see how this can be avoided. 
The problem resides on how `aur_rpc_utils::recursive_info` returns the info. It would be easier if for each dependency, it doesn't return a `Vec<String>` with the names of the deps and instead returned a `Vec<Package>` with a `Package` struct for each dependency.

Reply:
Hi, and sorry again for the late reply. I'm still pretty sure that the `recursive_info` can only be called once. The thing is that we get all the actually needed information from the very first call to this function, because as a result, we get `IndexMap<String, Package>`, so for each AUR package that is in the full dependency tree, we already know its `raur::Package` representation. This `Package` object already contains all the information that we need: make dependencies, check dependencies, normal dependencies, etc. It's only a question of accessing this data.

I tried to fix it locally, but it was non-trivial. See the branch https://github.com/vn971/rua/tree/deps_tree_limited and the latest commit in there https://github.com/vn971/rua/commit/5bda2ca4ec5b238729b90abf89f422ddedb7370e